gulpfile.js

```diff
@@ -538,21 +538,19 @@ function buildExtension(name, version, hasCss, options, opt_extraGlobs) {
       buildExtension(name, version, hasCss, copy);
     });
   }
+  let promise = Promise.resolve();
   if (hasCss) {
     mkdirSync('build');
     mkdirSync('build/css');
     const startTime = Date.now();
-    return buildExtensionCss(path, name, version, options).then(() => {
+    promise = buildExtensionCss(path, name, version, options).then(() => {
       endBuildStep('Recompiled CSS in', name, startTime);
-    }).then(function() {
-      if (options.compileOnlyCss) {
-        return Promise.resolve();
-      }
-      return buildExtensionJs(path, name, version, options);
     });
-  } else {
-    return buildExtensionJs(path, name, version, options);
+    if (options.compileOnlyCss) {
+      return promise;
+    }
   }
+  return promise.then(() => buildExtensionJs(path, name, version, options));
```

Comment:
This will end up compiling JS on css-only paths.

Reply:
As discussed, this is handled above in this function.